content/rendering-with-batch/batch/job_submission.py

```diff
@@ -0,0 +1,260 @@
+#
+# ##### END GPL LICENSE BLOCK #####
+def read_blend_rend_chunk(path):
+    """Extract from Blender's script library included in scripts/modules.
+    Reads the header of a blend file and returns scenes' information.
+
+    Keyword arguments:
+    path -- path where the blend file is located
+    """
+
+    import struct
+
+    blendfile = open(path, "rb")
+
+    head = blendfile.read(7)
+
+    if head[0:2] == b'\x1f\x8b':  # gzip magic
+        import gzip
+        blendfile.seek(0)
+        blendfile = gzip.open(blendfile, "rb")
+        head = blendfile.read(7)
+
+    if head != b'BLENDER':
+        print("not a blend file:", path)
+        blendfile.close()
+        return []
+
+    is_64_bit = (blendfile.read(1) == b'-')
+
+    # true for PPC, false for X86
+    is_big_endian = (blendfile.read(1) == b'V')
+
+    # Now read the bhead chunk!!!
+    blendfile.read(3)  # skip the version
+
+    scenes = []
+
+    sizeof_bhead = 24 if is_64_bit else 20
+
+    while blendfile.read(4) == b'REND':
+        sizeof_bhead_left = sizeof_bhead - 4
+
+        struct.unpack('>i' if is_big_endian else '<i', blendfile.read(4))[0]
+        sizeof_bhead_left -= 4
+
+        # We don't care about the rest of the bhead struct
+        blendfile.read(sizeof_bhead_left)
+
+        # Now we want the scene name, start and end frame. this is 32bites long
+        start_frame, end_frame = struct.unpack('>2i' if is_big_endian else '<2i', blendfile.read(8))
+
+        scene_name = blendfile.read(64)
+
+        scene_name = scene_name[:scene_name.index(b'\0')]
+
+        try:
+            scene_name = str(scene_name, "utf8")
+        except TypeError:
+            pass
+
+        scenes.append((start_frame, end_frame, scene_name))
+
+    blendfile.close()
+
+    return scenes
+
+def parse_arguments():
+    """Parses the command line arguments and stores the values in global variables.
+    """
+
+    parser = argparse.ArgumentParser(description='Submit an AWS Batch job that will render a Blender file in a distributed fashion.')
+    parser.add_argument('-i', dest='input_uri', type=str, required=True, help='S3 URI where the blender file is located')
+    parser.add_argument('-o', dest='output_uri', type=str, required=True, help='S3 URI where to upload the rendered file')
+    parser.add_argument('-f', dest='f_per_job', type=int, required=True, help='Number of frames that each job has to render')
+    parser.add_argument('-n', dest='job_name', type=str, required=True, help='Name of the job that will be submitted to Batch')
+    parser.add_argument('-q', dest='job_queue', type=str, required=True, help='Queue to which the job is submitted')
+    parser.add_argument('-d', dest='job_definition', type=str, required=True, help='Job definition used by the submitted job')
+    args = parser.parse_args()
+
+    if args.f_per_job < 1:
+        print('F_PER_JOB must be a positive integer')
+        sys.exit(2)
+
+    global INPUT_URI
```

Comment:
Yes, all global vars are declared at the top. Here I'm referencing the variable defined outside of the scope where that code is being executed, otherwise the value of the global vars won't change.